### DISCUSSION TITLE
Vercel App URL just downloads the python file

### DISCUSSION THREAD

**Discussion by scooter7** [vercel/community#2960] [Help] [answered]
### Summary

Hi,

When trying to visit https://chatchain-sg7vuezrf-scooter7.vercel.app/ , the index.py file just downloads on my local machine and doesn't actually run the app.

### Example

https://chatchain-sg7vuezrf-scooter7.vercel.app/

### Steps to Reproduce

When trying to visit https://chatchain-sg7vuezrf-scooter7.vercel.app/ , the index.py file just downloads on my local machine and doesn't actually run the app.

**Comment by amyegan** [comment]
Hi @scooter7. If you share your public repo or a minimal reproduction, I try to help you debug this

**Reply by scooter7** [reply]
Hi Amy,

Thanks so much! The repo lives at: https://github.com/scooter7/chatchain

**Comment by scooter7** [comment]
Hi @amyegan , I wonder if I need to add a requirements.txt file in the repo and/or update the toml with environment variables at https://github.com/scooter7/chatchain/blob/main/.chainlit/config.toml ?

Thanks again!

**Comment by amyegan** [comment, marked as answer]
Thanks for sharing the repo! It looks like you need to make a slight alteration to the project to get it working with [Vercel's Python Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python). 

I recommend creating an `api` folder in the root of your project and moving `index.py` into that folder. Then it will automatically be set up to run in a serverless function. 

You can use a `vercel.json` [rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) config to make the site go to the `/api` route when visiting the root domain.

```json
{
  "rewrites": [{ "source": "/", "destination": "/api" }]
}
```

**Comment by scooter7** [comment]
Hi @amyegan ,

Thanks so much for your help!

Once I create the vercel.json file, where should I put it in the Github file directory?

I was also assuming that I'll need somewhere within Vercel to store my openai key and Github tokens so that I can reference them within the index.py file?

Thanks again!

**Reply by amyegan** [reply]
Keys and tokens should be stored as [Environment Variables](https://vercel.com/docs/concepts/projects/environment-variables#) 🙂
